File: NetworkScan.cpp
#include "NetworkScan.h"

NetworkScan::NetworkScan(QObject *parent)
    : QObject(parent)
{

}

QList<QHostAddress> NetworkScan::start() const
{
    auto localIpAddress = QNetworkInterface::allAddresses();
    QList<QHostAddress> localIPv4 {};

    for (auto &ip : localIpAddress)
    {
        if (ip.protocol() == QAbstractSocket::IPv4Protocol)
            localIPv4.append(ip);
    }

    return localIPv4;
}

QList<QHostAddress> NetworkScan::refresh() const
{
    return this->start();
}

File: YeelightControl4PySide2/NetworkScan.py
from Modules import *


class NetworkScan(QObject):
    def __init__(self):
        QObject.__init__(self)
        
    def start(self):
        localIpAddress = QNetworkInterface.allAddresses()
        ips = list()
        ips.clear()
        for ip in localIpAddress:
            if ip.protocol() is QAbstractSocket.IPv4Protocol:
                ips.append(ip.toString())
        return ips
    
    def refresh(self):
        return self.start()